src/Payments.sol

```diff
@@ -296,24 +424,47 @@ contract Payments is
         Account storage payer = accounts[rail.token][rail.from];
         Account storage payee = accounts[rail.token][rail.to];
 
+        uint256 oldRate = rail.paymentRate;
+
+        // For terminated rails, can only reduce rate, not increase it
+        if (rail.terminationEpoch > 0) {
+            require(
+                newRate <= oldRate,
+                "failed because terminated at epoch: cannot increase rate on terminated rail"
+            );
+            require(
+                oneTimePayment <= rail.lockupFixed,
+                "failed because terminated at epoch: one-time payment exceeds fixed lockup"
+            );
+        }
+
         OperatorApproval storage approval = operatorApprovals[rail.token][
             rail.from
         ][rail.operator];
-        require(approval.isApproved, "Operator not approved");
 
-        // Settle the payer's lockup.
-        // If we're changing the rate, we require full settlement.
+        // Settle the payer's lockup to account for elapsed time
         (bool fullySettled, uint256 lockupSettledUpto) = settleAccountLockup(
             payer
         );
-        if (newRate != rail.paymentRate) {
-            require(
-                fullySettled && lockupSettledUpto == block.number,
-                "account lockup not fully settled; cannot change rate"
-            );
-        }
 
-        uint256 oldRate = rail.paymentRate;
+        // If we're increasing the rate, we require full account lockup setlement before we can proceed
+        require(
+            newRate <= oldRate ||
+                (fullySettled && lockupSettledUpto == block.number),
+            "account lockup not fully settled; cannot increase rate"
+        );
+
+        // We just settled the account lockup. If the client is still not able to guaruantee
+        // enough funds to lock up for the rail upto and including the current epoch, it means that the rail is in "debt"
+        // i.e. the client does not have enough funds to pay for services already taken on this rail.
+        // In this case, we should not allow rate modifications at all.
+        // The client should first deposit enough funds to pay for this "debt" and then the rate can be changed.
+        require(
+            newRate == oldRate ||
+                block.number < payer.lockupLastSettledAt + rail.lockupPeriod,
```

Comment:
```suggestion
                block.number <= payer.lockupLastSettledAt + rail.lockupPeriod,
```

I think?

Reply:
Yeah, you are right. a rail is in debt only if it is not able to pay upto and including the current epoch.